examples/general/non_local_effect_before_error_return/src/lib.rs

```diff
@@ -223,7 +224,8 @@ fn is_call_with_mut_ref<'tcx>(
         if locals.iter().any(|local| is_mut_ref_arg(mir, local))
             || constants.iter().any(|constant| is_const_ref(constant));
         then {
-            Some(*fn_span)
+            let func_string = format!("{func:#?}");
```

Comment:
It feels odd to me to convert something to a string using `Debug` formatting, only to later print it with `Display` formatting.

Could we instead change the return type to `Option<(&'a Operand, Span)>`, and include the operand directly in the warning using `Debug` formatting?

Also, I'm not certain, but I think just `{:?}` might work (i.e., instead of `{:#?}`).